Virtualenv module

mcyprian opened the pull request:
It is possible to get more accurate metadata using this module, it installs package to virtualenv in temporary directory and scans scripts,  and site_pacakges file that appears after instalation. It can also get list of other installed packages - package runtime dependancies. Metadata from virtualenv are appended to metadata from archive and don't overwrite them.

Commit:
Add virtualenv module to extract more metadata

Commit:
Make possible to append venv metadata to already extracted archive metadata

Commit:
Improve error handling in virtualvenv module and updating of runtime_deps

sYnfo commented on pyp2rpm/virtualenv.py:
```diff
@@ -0,0 +1,159 @@
+        Creates new virtualenv in temp_dir
+        '''
+        proc = Popen(['virtualenv', '-p', VENV_INTERPRETER, self.temp_dir + '/venv/'],
+                     stdout=DEVNULL, stderr=PIPE)
```
Have you considered using the virtualenv-api package? It seems it would be a good fit for this usecase.

[0] https://pypi.python.org/pypi/virtualenv-api/

Commit:
Use virtualenv-api instead of subprocesses

Commit:
Use tempfile.mkdtemp to make code Python2 compatible

Commit:
Change travis to run tests of all Python versions properly

Commit:
Change virtualenv module to be Python2.6 compatible

Commit:
Add tests for virtualenv module

Commit:
Change test_virtualenv module to be Python2.6 compatible

Commit:
Complete tests for virtualenv module

mcyprian commented:
I 've rewritten it using virtualenv-api, code doesn't seem to be much faster but it is definitely a bit nicer. I 've also added tests for this module and made it all Python2.6 compatible.

Commit:
Fix pylint warnings

Commit:
Update CHANGELOG

Commit:
Improve indication if pacakge should be taken from pypi of local file, fixes #20

rkuska commented:
Trying `pyp2rpm pytest` I end up with:

```
%{_bindir}/py.test
%{_bindir}/py.test-2
%{_bindir}/py.test-%{python2_version}
%{_bindir}/py.test-2.7
%{_bindir}/py.test-2.7-2
%{_bindir}/py.test-2.7-%{python2_version}
```

Commit:
Overwrite scripts containing -MAJOR.MINOR suffix
- remove scripts containing suffixes from metadata
- make simlinks having -MAJOR.MINOR suffix when generating spec
- lowercasing PyPI names during venv requires extraction

Commit:
Append runtime deps to build deps

Commit:
Move default BuildRequires to metadata initialization,
Add scripts filter to virtualenv

Commit:
Use base python interpreter in venv
